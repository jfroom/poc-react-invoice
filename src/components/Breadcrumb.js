import React from 'react';
import {LinkContainer} from 'react-router-bootstrap';

const Breadcrumb = () => (
  <div>
    <ol className="breadcrumb">
      <li><LinkContainer to='/'><a>Invoices</a></LinkContainer></li>
      <li><LinkContainer to='/new'><a>New Invoice</a></LinkContainer></li>
    </ol>
  </div>
)

export default Breadcrumb
