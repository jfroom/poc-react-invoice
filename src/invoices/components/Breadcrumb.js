// @flow
import React, { Component } from 'react'
import { LinkContainer } from 'react-router-bootstrap'
import type { Location } from '../models'
import styled from 'styled-components'

type BreadcrumbLink = {
  text: string,
  to?: string,
  active?: boolean,
}

const StyleWrap = styled.div`
  text-transform: capitalize;
`

class Breadcrumb extends Component {
  static defaultProps = {
    location: {
      pathname: ''
    }
  }
  props: {
    location: Location
  }
  render () {
    const { location } = this.props
    const items:Array<BreadcrumbLink> = [{ text: 'Invoices', to: '/', active: true }]
    if (location.pathname !== '/') {
      const action = location.pathname.split('/')[1]
      items.push({ text: `${action} Invoice` })
    }
    items[items.length - 1].active = false

    const getInner = (link) => {
      if (link.active) {
        return (
          <LinkContainer to={link.to} isActive={() => link.active}>
            <a>{link.text}</a>
          </LinkContainer>
        )
      }
      return <span>{link.text}</span>
    }

    return (
      <StyleWrap>
        <ol className="breadcrumb">
          {items.map((item, idx) => (<li key={idx}>{getInner(item)}</li>))}
        </ol>
      </StyleWrap>
    )
  }
}

export default Breadcrumb
