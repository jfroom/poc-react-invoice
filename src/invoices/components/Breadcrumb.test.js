import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Breadcrumb from './Breadcrumb'
import { BrowserRouter as Router } from 'react-router-dom'

it('Invoices', () => {
  // Mount & verify snapshot
  const props = { location: { pathname: '/' }}
  const wrap = mount(<Router><Breadcrumb {...props} /></Router>)
  expect(toJson(wrap)).toMatchSnapshot()
})

it('Invoices / New Invoice', () => {
  // Mount & verify snapshot
  const props = { location: { pathname: '/new' }}
  const wrap = mount(<Router><Breadcrumb {...props} /></Router>)
  expect(toJson(wrap)).toMatchSnapshot()
})

it('Invoices / Edit Invoice', () => {
  // Mount & verify snapshot
  const props = { location: { pathname: '/edit/1' }}
  const wrap = mount(<Router><Breadcrumb {...props} /></Router>)
  expect(toJson(wrap)).toMatchSnapshot()
})
