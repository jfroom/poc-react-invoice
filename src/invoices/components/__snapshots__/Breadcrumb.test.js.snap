// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoices / Edit Invoice 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Breadcrumb
      location={
        Object {
          "pathname": "/edit/1",
        }
      }
    >
      <styled.div>
        <div
          className="sc-bdVaJa hhwEPU"
        >
          <ol
            className="breadcrumb"
          >
            <li>
              <LinkContainer
                activeClassName="active"
                exact={false}
                isActive={[Function]}
                replace={false}
                strict={false}
                to="/"
              >
                <Route
                  exact={false}
                  path="/"
                  strict={false}
                >
                  <a
                    className=" active"
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Invoices
                  </a>
                </Route>
              </LinkContainer>
            </li>
            <li>
              <span>
                edit Invoice
              </span>
            </li>
          </ol>
        </div>
      </styled.div>
    </Breadcrumb>
  </Router>
</BrowserRouter>
`;

exports[`Invoices / New Invoice 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Breadcrumb
      location={
        Object {
          "pathname": "/new",
        }
      }
    >
      <styled.div>
        <div
          className="sc-bdVaJa hhwEPU"
        >
          <ol
            className="breadcrumb"
          >
            <li>
              <LinkContainer
                activeClassName="active"
                exact={false}
                isActive={[Function]}
                replace={false}
                strict={false}
                to="/"
              >
                <Route
                  exact={false}
                  path="/"
                  strict={false}
                >
                  <a
                    className=" active"
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Invoices
                  </a>
                </Route>
              </LinkContainer>
            </li>
            <li>
              <span>
                new Invoice
              </span>
            </li>
          </ol>
        </div>
      </styled.div>
    </Breadcrumb>
  </Router>
</BrowserRouter>
`;

exports[`Invoices 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Breadcrumb
      location={
        Object {
          "pathname": "/",
        }
      }
    >
      <styled.div>
        <div
          className="sc-bdVaJa hhwEPU"
        >
          <ol
            className="breadcrumb"
          >
            <li>
              <span>
                Invoices
              </span>
            </li>
          </ol>
        </div>
      </styled.div>
    </Breadcrumb>
  </Router>
</BrowserRouter>
`;
